### ISSUE TITLE
Bug: namespaced constants conflict with toplevel constants

### ISSUE THREAD

**Issue by hachi8833** [goby-lang/goby#500]
When a namespaced constant (`Foo::VAL`) is defined first and then a toplevel constant with the same name (`VAL`), it works.

But when a toplevel constant is defined first and then a namespaced constant with the same name (`Bar::VAL`) is defined, the constants are conflicted.

```
class Foo
  VAL="Foo's constant value"
  def foo
    VAL
  end
end

VAL="toplevel constant value"

VAL              #» toplevel constant value
Foo::VAL         #» Foo's constant value
Foo.new.foo      #» Foo's constant value

class Bar
  VAL="Bar's constant value"
  def bar
    VAL
  end
end   #» ERROR: ConstantAlreadyInitializedError: Constant VAL already been initialized. Can't assign value to a constant twice.. At REPL:2
```

This works on Ruby.

**Comment by st0012** [comment]
@hachi8833 Can you write a simple test script and expected outcome this bug?

**Comment by hachi8833** [comment]
Here you go:

```go
// Checks if constants are namespaced
func TestConstantNamespace(t *testing.T) {
	tests := []struct {
		input    string
		expected []interface{}
	}{
		{`
		class Foo
			VAL="Foo's constant value"
			def foo
 				VAL
 			end
 		end
 		VAL="toplevel constant value"
		a = [VAL, Foo::VAL, Foo.new.foo]
		a
		`, []interface{}{"toplevel constant value", "Foo's constant value", "Foo's constant value"}},
		{`
		VAL="toplevel constant value"
		class Foo
		 VAL="Foo's constant value"
		 def foo
		   VAL
		 end
		end
		a = [VAL, Foo::VAL, Foo.new.foo]
		a
		`, []interface{}{"toplevel constant value", "Foo's constant value", "Foo's constant value"}},
	}

	for i, tt := range tests {
		vm := initTestVM()
		evaluated := vm.testEval(t, tt.input, getFilename())
		testArrayObject(t, i, evaluated, tt.expected)
		vm.checkCFP(t, i, 0)
		vm.checkSP(t, i, 1)
	}
}
```

**Comment by hachi8833** [comment]
FYI: Ruby ~2.4 has the similar issue https://bugs.ruby-lang.org/issues/11547 and it has been fixed in 2.5.
